src/site/pages/digital/src/containers/ExprToCircuitPopup/index.tsx

```diff
@@ -0,0 +1,201 @@
+                   close={() => dispatch(CloseHeaderPopups())}>
+                <div className="exprtocircuit__popup">
+                    { errorMessage && <p className="exprtocircuit__popup__errorMessage">{"ERROR: " + errorMessage}</p> }
+                    <input title="Enter Circuit Expression" type="text"
+                               value={expression}
+                               placeholder="!a | (B^third)"
+                               onChange={e => setExpression(e.target.value)} />
+                    <br/>
+
+                    <div className="exprtocircuit__popup__settings">
+                        <div>
+                            <h3>Notation</h3>
+                            {formats}
+                            <div>
+                                <input type="radio" name="format" checked={format === "custom"} onChange={() => setFormat("custom")} value={"custom"} />
+                                <label htmlFor={"custom"}>Custom</label><br/>
+                            </div>
+                            {
+                                format === "custom" &&
+                                <div>
+                                    {customOpsInput}
+                                </div>
+                            }
+                        </div>
+
+                        <div>
+                            <h3>Options</h3>
+                            <input onChange={() => setIsIC(!isIC)} checked={isIC} type="checkbox" id="isIC" name="isIC" />
+                            <label htmlFor="isIC">Generate into IC</label>
+                            <br/>
+                            <br/>
+                            <label>Input Component Type:  </label>
+                            <select id="input"
+                                value={input}
+                                onChange={e => setInput(e.target.value)}
+                                onBlur={e => setInput(e.target.value)}>
+                            {inputTypes}
+                            </select>
+                        </div>
+
+                    </div>
+
+                    <button className="exprtocircuit__popup__generate" type="button" disabled={expression===""} onClick={() => {
+                        try {
+                            generate(mainInfo, expression, isIC, input, format, new Map(Object.entries(userOps) as [FormatLabels, string][]));
+                            setExpression("");
+                            setErrorMessage("");
+                            dispatch(CloseHeaderPopups());
+                        }
+                        catch (err) {
+                            setErrorMessage(err.message);
+                        }
+                    }}>Generate</button>
+
+                    <button className="cancel" type="button" onClick={() => {
+                        setExpression("");
+                        setErrorMessage("");
+                        dispatch(CloseHeaderPopups());
```

Comment:
You can declare like a `reset` function above (in the component still though) since this is being used twice